lib/grizzly/zwave/commands/switch_multilevel_set.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Grizzly.ZWave.Commands.SwitchMultilevelSet do
+  @moduledoc """
+  What does this command do??
+
+  Params:
+
+    * `:target_value` - '`:off`, `:previous` or a value betweem 1 and 99
+
+    * `:duration` - How long the switch should take to reach target value, 0 -> instantly, 1..127 -> seconds, 128..254 -> minutes, 256 -> factory default (option v2)
+
+  """
+
+  @behaviour Grizzly.ZWave.Command
+
+  alias Grizzly.ZWave.{Command, DecodeError}
+  alias Grizzly.ZWave.CommandClasses.SwitchMultilevel
+
+  @type param :: {:target_value, non_neg_integer()} | {:duration, non_neg_integer()}
+
+  @impl true
+  @spec new([param()]) :: {:ok, Command.t()}
+  def new(params) do
+    # TODO: validate opts
+    command = %Command{
+      name: :switch_multilevel_set,
+      command_byte: 0x01,
+      command_class: SwitchMultilevel,
+      params: params,
+      impl: __MODULE__
+    }
+
+    {:ok, command}
+  end
+
+  @impl true
+  def encode_params(command) do
+    target_value_byte = encode_target_value(Command.param!(command, :target_value))
+
+    case Command.param(command, :duration) do
+      nil ->
+        <<target_value_byte>>
+
+      duration_byte ->
+        <<target_value_byte, duration_byte>>
+    end
+  end
+
+  def encode_target_value(:off), do: 0x00
+  def encode_target_value(target_value) when target_value in 0..99, do: target_value
+  def encode_target_value(:previous), do: 0xFF
+
+  @impl true
+  @spec decode_params(<<_::8, _::_*8>>) ::
```

Comment:
I like the addition of the type spec, but can make the `<<_::8, _::_*8>>` type just be `binary()`, they mean the same thing but `binary()` is just easier to read and reason about.

Reply:
I removed the spec for now to be consistent.